Move to ws folder for rosdep command, fixes #876

dave992 opened the pull request:
Executing the `rosdep` command from the root of the repository results in multiple packages found with the same name when run from the CI on GitLab. (See #876 for more information)

I have been unable to find the root cause for this, but this fix seems to circumvent the issue by restricting the folders `rosdep` can search in. I believe functionally this should behave the same.

mathias-luedtke commented on industrial_ci/src/workspace.sh:
```diff
@@ -373,7 +373,7 @@ function ici_install_dependencies {
     fi
 
-    ROS_PACKAGE_PATH="$cmake_prefix_path${ROS_PACKAGE_PATH:-}" ici_cmd ici_quiet ici_filter "(executing command)|(Setting up)" ici_exec_in_workspace "$extend" "." rosdep install "${rosdep_opts[@]}"
+    ROS_PACKAGE_PATH="$cmake_prefix_path${ROS_PACKAGE_PATH:-}" ici_cmd ici_quiet ici_filter "(executing command)|(Setting up)" ici_exec_in_workspace "$extend" "$@" rosdep install "${rosdep_opts[@]}"
```
This will only work, if only one argument is passed.

I am not sure, why this fixes the issue.
This just changes the execution directory and not the discovery of packages.

mathias-luedtke replied:
```suggestion
    ROS_PACKAGE_PATH="$cmake_prefix_path${ROS_PACKAGE_PATH:-}" ici_cmd ici_quiet ici_filter "(executing command)|(Setting up)" ici_exec_in_workspace "$extend" "${1:-.}" rosdep install "${rosdep_opts[@]}"
```

Should do the trick.
I still have to figure out why rosdep behaves that way

dave992 committed:
Move to ws folder for rosdep command, fixes #876

Executing the rosdep command from the root of repository (!= workspace),
results in multiple packages found with the same name when run from the
CI on GitLab.

Unable to reproduce this from inside the same container as the CI runs,
but this fix seems to circumvent the issue by restricting the folders
rosdep can search in. Functionally this should behave the same.
